models/members.js

```diff
@@ -5,21 +5,25 @@
 
 const firestore = require("../utils/firestore");
 const userModel = firestore.collection("users");
-
+const { ROLES } = require("../constants/roles");
 /**
  * Fetches the data about our users
  * @return {Promise<userModel|Array>}
  */
 
-const fetchUsers = async () => {
+const fetchUsers = async (queryParams = {}) => {
   try {
     const snapshot = await userModel.get();
-
     const allMembers = [];
 
+    let { includeArchived } = queryParams;
+    if (includeArchived === "true") includeArchived = true;
+    else includeArchived = false;
+
     if (!snapshot.empty) {
       snapshot.forEach((doc) => {
         const memberData = doc.data();
+        if (!includeArchived && memberData?.roles && memberData.roles[ROLES.ARCHIVED] === true) return;
```

Comment:
Instead of fetching everything and then removing archived users, can we write a firestore query such that we only fetch non-archived users in the first place.
Imagine if there are 100 non-archived users and 10000 archived users. We would have paid for 10100 calls, and used only a small portion of it.
Every DB/firestore read costs us.

Reply:
Even with index we can't query for unarchived users if roles or archived field is absent https://firebase.google.com/docs/firestore/query-data/queries#limitations

Also there's an index for map fields by default
https://firebase.google.com/docs/firestore/query-data/index-overview#automatic_indexing

Reply:
Resolves #651

Reply:
What is the full form of DNE hear?

Reply:
> What is the full form of DNE hear?

DNE - does not exist